### PR TITLE
picketlink-sts

### PR DESCRIPTION
**Pull request by pskopek**
Adding picketlink-sts to quckstarts to master branch.
https://bugzilla.redhat.com/show_bug.cgi?id=1011975

### CODE REVIEW

**Comment by rafabene** [comment]
What's the content difference between #675 PR ? Just the destination branch ?

File: picketlink-sts/README.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+======================================================
+Author: Peter Skopek
+Level: Advanced
+Technologies: WS-Trust, SAML
+Summary: This project is an implementation of a WS-Trust Security Token Service.
+Target Product: EAP
+Product Versions: EAP 6.1, EAP 6.2
+Source: <https://github.com/jboss-developer/jboss-eap-quickstarts/>
+ 
+
```
Comment (sgilda) [inline picketlink-sts/README.md:11]: This is missing the 'What is it?' section that describes the quickstart.

File: picketlink-sts/README.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+If you have not yet done so, you must [Configure Maven](../README.md#configure-maven) before testing the quickstarts.
+
+
+Additional PicketLink STS configuration options
+-----------------------------------------------
+
+Application is preconfigured to use "picketlink-sts" security domain with user.properties and roles.properties files.
+This is not suitable for production use. Change application security according to your organization standards.
+
+For more information about PicketLink [see:](http://docs.jboss.org/picketlink/2/2.1.7.Final/reference/html/)
```
Comment (sgilda) [inline picketlink-sts/README.md:32]: This link renders strangely. You might want to say something like: 

For more information about PicketLink see the [PicketLink Reference Documentation](http://docs.jboss.org/picketlink/2/2.1.7.Final/reference/html/).

File: picketlink-sts/README.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+        mvn clean install jboss-as:deploy
+4. This will deploy `target/picketlink-sts.war` to the running instance of the server.
+
+
+Further actions
+---------------
+
+This quickstart is just a service application for another applications you can use. See more examples in [PicketLink project documemntation.](http://docs.jboss.org/picketlink/2/2.1.7.Final/reference/html/ch01.html#sid-819345). You can take a look for more applications to play with at [PicketLink Quickstarts.](https://docs.jboss.org/author/display/PLINK/PicketLink+Quickstarts)
+
```
Comment (sgilda) [inline picketlink-sts/README.md:67]: The readme needs to describe how to undeploy the application and remove the configuration.

File: picketlink-sts/README.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+Application is preconfigured to use "picketlink-sts" security domain with user.properties and roles.properties files.
+This is not suitable for production use. Change application security according to your organization standards.
+
+For more information about PicketLink [see:](http://docs.jboss.org/picketlink/2/2.1.7.Final/reference/html/) 
+
+
+Configure the Security Domain Using the JBoss CLI 
+-------------------------------------------------
+
+1. Start the JBoss server by typing the following:
```
Comment (sgilda) [inline picketlink-sts/README.md:38]: The first step should be to back up the current configuration.

**Comment by pedroigor** [comment]
@sgilda Below is the "What is it?" section for this quickstart:

This example demonstrates how to deploy a fully compliant WS-Trust Security Token Service. 

WS-Trust extends the WS-Security specification to allow issuing, renewing, and validation of security tokens. A lot of what WS-Trust does centers around the use of a "Security Token Service", or STS. The STS is contacted to obtain security tokens that are used to create messages to talk to the services. The primary use of the STS is to acquire SAML tokens used to talk to the service.

The STS plays an important role when you need to propagate credentials between different layers, for example, the web and service layer.

Basically, this application is a JAX-WS Endpoint based on PicketLink's WS-Trust implementation. Which by default allows you to issue, renew and validate SAML assertions.

PicketLink also supports different token providers, which means you can provide your own custom security tokens.

The PicketLink STS documentation is available [here](https://docs.jboss.org/author/display/PLINK/Security+Token+Server+%28STS%29).

**Comment by sgilda** [comment]
I have modified this to go against master. I've enhanced the README file. Now I'm stuck at the "Access the Application"

When I access http://localhost:8080/jboss-picketlink-sts, I get a dialog:

The server http:localhost:8080 requires a username and password. The server says PicketLinkSTSRealm.

Do you need to add users for this application? Or is something hard-coded? What do I enter for the user/password to gain access?

**Comment by sgilda** [comment]
I see users in the users.properties file but none of them seem to work. 

We need to provide instructions for after the app is deployed.

**Comment by sgilda** [comment]
Closing this pull. It was replaced by https://github.com/jboss-developer/jboss-eap-quickstarts/pull/683
